### PR TITLE
IB-3313 Added /group_document_folders EP

### PR DESCRIPTION
**Pull request by sam-ludlow**
https://io1dev.atlassian.net/browse/IB-3313

### CODE REVIEW

File: api.swagger.yaml [hunk 1]
```diff
@@ -3998,6 +3998,149 @@ paths:
...
+          description: 'Filter by name. Matches on whole words.'
+          in: query
+          schema:
+            type: string
+        - description: 'Filter by external ID.'
+          in: query
+          name: external_id
+          schema:
+            type: string
+        - name: status
```
Comment (ParryLuke) [inline api.swagger.yaml]: 'status' and 'published' should be removed from all group doc folder endpoints as there is no way to change this in the product.

The value will exist in solr/services but as it is not exposed we can assume that all group doc folders will be published (we should still send published: true to services and filter published: true silently between node and solr but just not expose it).

File: api.swagger.yaml [hunk 2]
```diff
@@ -8274,6 +8417,74 @@ components:
...
+              example: '01234567-89ab-cdef-1234-56789abcdef0'
+            has_children:
+              description: 'Indicates that this group document folder has children.'
+              type: boolean
+              example: true
+            group:
+              description: 'The group of this document folder.'
+              type: object
+              allOf:
+                - $ref: '#/components/schemas/group-schema'
```
[the remark this answers is not in the document]
Comment (sam-ludlow) [reply api.swagger.yaml]: done

File: api.swagger.yaml [hunk 3]
```diff
@@ -8274,6 +8417,74 @@ components:
...
+        - $ref: '#/components/schemas/group-document-folder-schema-common'
+    group-document-folder-schema-update:
+      allOf:
+        - $ref: '#/components/schemas/group-document-folder-schema-common'
+        - type: object
+          properties:
+            group_uuid:
```
[the remark this answers is not in the document]
Comment (sam-ludlow) [reply api.swagger.yaml]: done

File: api.swagger.yaml (hunk 1 above)
Comment (sam-ludlow) [reply api.swagger.yaml]: done, took it out of the /list filter also.

**Comment by sam-ludlow** [comment]
Moved these EPs to within /groups/uuid so doing a new PR

**Pull request closed without merging**